velox/exec/prefixsort/tests/PrefixEncoderTest.cpp

```diff
@@ -0,0 +1,296 @@
+          encoder.isAscending(),
+          false,
+          CompareFlags::NullHandlingMode::kNullAsValue};
+      const auto leftVector = fuzzer.fuzzFlat(type, vectorSize);
+      const auto rightVector = fuzzer.fuzzFlat(type, vectorSize);
+      const auto* left = leftVector->values()->as<NativeType>();
+      const auto* right = rightVector->values()->as<NativeType>();
+
+      char leftEncoded[sizeof(NativeType) + 1];
+      char rightEncoded[sizeof(NativeType) + 1];
+
+      for (auto i = 0; i < vectorSize; ++i) {
+        const auto leftValue = leftVector->isNullAt(i)
+            ? std::nullopt
+            : std::optional<NativeType>(*(left + i));
+        const auto rightValue = rightVector->isNullAt(i)
+            ? std::nullopt
+            : std::optional<NativeType>(*(right + i));
+        encoder.encode(leftValue, leftEncoded);
+        encoder.encode(rightValue, rightEncoded);
+
+        const auto result = compare(leftEncoded, rightEncoded);
+        const auto expected =
+            leftVector->compare(rightVector.get(), i, i, compareFlag).value();
+        ASSERT_FALSE((result ^ expected) >> 31);
+        if (expected == 0) {
+          ASSERT_EQ(result, 0);
+        }
+      }
+    };
+
+    test(ascNullsFirstEncoder_);
+    test(ascNullsLastEncoder_);
+    test(descNullsFirstEncoder_);
+    test(descNullsLastEncoder_);
+  };
+
+ protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance({});
+  }
+
+ private:
+  const PrefixSortEncoder ascNullsFirstEncoder_ = {true, true};
+  const PrefixSortEncoder ascNullsLastEncoder_ = {true, false};
+  const PrefixSortEncoder descNullsFirstEncoder_ = {false, true};
+  const PrefixSortEncoder descNullsLastEncoder_ = {false, false};
+};
+
+TEST_F(PrefixEncoderTest, encode) {
+  {
+    uint64_t ascExpected = 0x8877665544332211;
+    uint64_t descExpected = 0x778899aabbccddee;
+    testEncode<uint64_t>(
+        0x1122334455667788, (char*)&ascExpected, (char*)&descExpected);
+  }
+
+  {
+    int64_t ascExpected = 0x8877665544332291;
+    int64_t descExpected = 0x778899aabbccdd6e;
+    testEncode<int64_t>(
+        0x1122334455667788, (char*)&ascExpected, (char*)&descExpected);
+  }
+  {
+    uint32_t ascExpected = 0x44332211;
+    uint32_t descExpected = 0xbbccddee;
+    testEncode<uint32_t>(0x11223344, (char*)&ascExpected, (char*)&descExpected);
+  }
+  {
+    int32_t ascExpected = 0x44332291;
+    int32_t descExpected = 0xbbccdd6e;
+    testEncode<int32_t>(0x11223344, (char*)&ascExpected, (char*)&descExpected);
+  }
+
+  {
+    uint32_t ascExpected = 0x0050c3c7;
+    uint32_t descExpected = 0xffaf3c38;
+    testEncode<float>(100000.00, (char*)&ascExpected, (char*)&descExpected);
+  }
+
+  {
+    uint64_t ascExpected = 0x00000000006af8c0;
+    uint64_t descExpected = 0xffffffffff95073f;
+    testEncode<double>(100000.00, (char*)&ascExpected, (char*)&descExpected);
+  }
+}
+
+TEST_F(PrefixEncoderTest, compare) {
+  testCompare<uint64_t>();
```

Comment:
1.For encode<uintxx_t>,  I check the types of Velox supported now,  uint64_t would be used for Timestamp`s nanos_.  I think the logic of encode Timestamp could be put out side of Encoder, because it will call 'encode' method tiwce for second & nanos_.

The uint32_t type maybe not used now.

2. the encodeNoNulls<uint32_t> is called both by encodeNoNulls<int32_t> , encodeNoNulls<float>

Reply:
I see. We may want to add explicit support for Timestamp since we'd want to encode null-flag only once per Timestamp values, not twice (once for seconds and once again for nanos).